packages/alchemy/src/react/components/button.tsx

```diff
@@ -13,6 +13,7 @@ type ButtonProps = (
     }
   | { type: "secondary"; icon?: never }
   | { type: "social"; icon?: string | ReactNode }
+  | { type: "link"; icon?: never }
```

Comment:
(Nit) If you want, you could probably reduce some repetition in this component. For one thing, you could put the union inside the `type` field for the non-icons
```ts
type ButtonProps = (
  | { type?: "primary" | "secondary" | "link"; icon?: never }
  | { type: "social"; icon?: string | ReactNode }
) // ...
```
and then instead of the cases below you could have the component once and do cases for the class name:
```ts
const btnClass = type === "secondary"
  ? "btn-secondary"
  : type === "social"
  ? "btn-auth"
  : type === link
  ? "btn-link"
  : "btn-primary";
return (
  <button className={`btn ${btnClass}`} {...props} ref={ref}>
    {icon && <span>{icon}</span>}
    {children}
  </button>
);
```

Reply:
yea this is much nicer, gonna do this